flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/StatisticsValidator.java

```diff
@@ -0,0 +1,127 @@
+	public static final String STATISTICS_ROW_COUNT = "statistics.row-count";
+	public static final String STATISTICS_COLUMNS = "statistics.columns";
+
+	// per column properties
+	public static final String NAME = "name";
+	public static final String DISTINCT_COUNT = "distinct-count";
+	public static final String NULL_COUNT = "null-count";
+	public static final String AVG_LENGTH = "avg-length";
+	public static final String MAX_LENGTH = "max-length";
+	public static final String MAX_VALUE = "max-value";
+	public static final String MIN_VALUE = "min-value";
+
+	@Override
+	public void validate(DescriptorProperties properties) {
+		properties.validateInt(STATISTICS_PROPERTY_VERSION, true, 0, Integer.MAX_VALUE);
+		properties.validateLong(STATISTICS_ROW_COUNT, true, 0);
+		validateColumnStats(properties, STATISTICS_COLUMNS);
+	}
+
+	// utilities
+
+	public static Map<String, String> normalizeColumnStats(ColumnStats columnStats) {
+		Map<String, String> stats = new HashMap<>();
+		if (columnStats.getNdv() != null) {
+			stats.put(DISTINCT_COUNT, String.valueOf(columnStats.getNdv()));
+		}
+		if (columnStats.getNullCount() != null) {
+			stats.put(NULL_COUNT, String.valueOf(columnStats.getNullCount()));
+		}
+		if (columnStats.getAvgLen() != null) {
+			stats.put(AVG_LENGTH, String.valueOf(columnStats.getAvgLen()));
+		}
+		if (columnStats.getMaxLen() != null) {
+			stats.put(MAX_LENGTH, String.valueOf(columnStats.getMaxLen()));
+		}
+		if (columnStats.getMax() != null) {
+			stats.put(MAX_VALUE, String.valueOf(columnStats.getMax()));
+		}
+		if (columnStats.getMin() != null) {
+			stats.put(MIN_VALUE, String.valueOf(columnStats.getMin()));
+		}
+		return stats;
+	}
+
+	public static void validateColumnStats(DescriptorProperties properties, String key) {
+
+		// filter for number of columns
+		int columnCount = properties.getIndexedProperty(key, NAME).size();
+
+		for (int i = 0; i < columnCount; i++) {
+			properties.validateString(key + "." + i + "." + NAME, false, 1);
+			properties.validateLong(key + "." + i + "." + DISTINCT_COUNT, true, 0L);
+			properties.validateLong(key + "." + i + "." + NULL_COUNT, true, 0L);
+			properties.validateDouble(key + "." + i + "." + AVG_LENGTH, true, 0.0);
+			properties.validateInt(key + "." + i + "." + MAX_LENGTH, true, 0);
+			properties.validateDouble(key + "." + i + "." + MAX_VALUE, true, 0.0);
+			properties.validateDouble(key + "." + i + "." + MIN_VALUE, true, 0.0);
+		}
+	}
+
+	public static Map<String, ColumnStats> readColumnStats(DescriptorProperties properties, String key) {
+
+		// filter for number of columns
+		int columnCount = properties.getIndexedProperty(key, NAME).size();
+
+		Map<String, ColumnStats> stats = new HashMap<>();
+		for (int i = 0; i < columnCount; i++) {
+			final String propertyKey = key + "." + i + "." + NAME;
+			String name = properties.getOptionalString(propertyKey).orElseThrow(exceptionSupplier(propertyKey));
```

Comment:
If you use `getString` instead of `getOptionalString`, you don't need the custom exception supplier.